src/signature.rs

```diff
@@ -181,7 +182,9 @@ impl TryFrom<&ed25519::Signature> for InternalSignature {
 }
 
 impl From<InternalSignature> for ed25519::Signature {
+    #[allow(clippy::unwrap_used)]
     fn from(sig: InternalSignature) -> ed25519::Signature {
+        // This function is actually infallible but it's marked as fallible
```

Comment:
Just making sure. Is this true? [Source](https://github.com/RustCrypto/signatures/blob/0bb3f07a0677b4c87a1b580c81eac68bbfc69f1c/ed25519/src/lib.rs#L304-L311)

Reply:
The conversion won't ever fail because a serialized `InternalSignature` will never violate `ed25519::Signature`'s invariant that `s` does not overflow the curve's order.

However, if it were to, `ed25519::Signature::from_bytes` will fail.